crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -120,6 +123,215 @@ impl<'a> DataRef<'a> {
 // Type aliases for lock gaurds
 type SharedWriteGuard<T> = ArcRwLockWriteGuard<RawRwLock, T>;
 type SharedMutexGuard<T> = ArcMutexGuard<RawMutex, T>;
+type SharedReadGuard<T> = ArcRwLockReadGuard<RawRwLock, T>;
+
+#[allow(dead_code)]
+pub struct TxId {
+    committed_state_shared_lock: SharedReadGuard<CommittedState>,
+    lock_wait_time: Duration,
+    timer: Instant,
+}
+
+#[allow(dead_code)]
+impl TxId {
+    fn table_id_from_name(&self, table_name: &str, database_address: Address) -> super::Result<Option<TableId>> {
+        self.iter_by_col_eq(
+            &ExecutionContext::internal(database_address),
+            &ST_TABLES_ID,
+            NonEmpty::new(StTableFields::TableName.col_id()),
+            AlgebraicValue::String(table_name.to_owned()),
+        )
+        .map(|mut iter| {
+            iter.next()
+                .map(|row| TableId(*row.view().elements[0].as_u32().unwrap()))
+        })
+    }
+
+    fn iter<'a>(&'a self, ctx: &'a ExecutionContext, table_id: &TableId) -> super::Result<Iter<'a>> {
+        self.committed_state_shared_lock.iter(ctx, table_id)
+    }
+
+    fn table_exists(&self, table_id: &TableId) -> bool {
+        self.committed_state_shared_lock.table_exists(table_id)
+    }
+
+    /// Returns an iterator,
+    /// yielding every row in the table identified by `table_id`,
+    /// where the values of `cols` are contained in `range`.
+    fn iter_by_col_range<'a, R: RangeBounds<AlgebraicValue>>(
+        &'a self,
+        ctx: &'a ExecutionContext,
+        table_id: &TableId,
+        cols: NonEmpty<ColId>,
+        range: R,
+    ) -> super::Result<IterByColRange<'a, R>> {
+        match self.committed_state_shared_lock.index_seek(table_id, &cols, &range) {
+            Some(committed_rows) => Ok(IterByColRange::CommittedIndex(CommittedIndexIter {
+                ctx,
+                table_id: *table_id,
+                tx_state: None,
+                committed_state: &self.committed_state_shared_lock,
+                committed_rows,
+                num_committed_rows_fetched: 0,
+            })),
+            None => self
+                .committed_state_shared_lock
+                .iter_by_col_range(ctx, table_id, cols, range),
+        }
+    }
+
+    fn iter_by_col_eq<'a>(
+        &'a self,
+        ctx: &'a ExecutionContext,
+        table_id: &TableId,
+        cols: NonEmpty<ColId>,
+        value: AlgebraicValue,
+    ) -> super::Result<IterByColEq<'_>> {
+        self.iter_by_col_range(ctx, table_id, cols, value)
+    }
+
+    fn schema_for_table<'tx>(&'tx self, table_id: TableId) -> super::Result<Cow<'tx, TableSchema>> {
```

Comment:
This has been mostly copied from `CommittedState::schema_for_table`, as `iter_by_col_eq()` method implementation differs for `TxId` and `CommittedState`.
I am experimenting if we can avoid duplication by some way in next PR as it's dead-code right now.